File: leetcode/1491. Average Salary Excluding the Minimum and Maximum Salary/main.ex
defmodule Solution do
  @spec average(salary :: [integer]) :: float
  def average(salary) do
    max = Enum.max(salary)
    min = Enum.min(salary)
    sum = Enum.sum(salary)
    count = Enum.count(salary)

    (sum - max - min) / (count - 2)
  end
end

File: leetcode/1491. Average Salary Excluding the Minimum and Maximum Salary/main.rs
impl Solution {
    pub fn average(salary: Vec<i32>) -> f64 {
        let mut max = i32::MIN;
        let mut min = i32::MAX;
        let mut sum = 0;
        let len = salary.len();

        for item in salary {
            max = std::cmp::max(item, max);
            min = std::cmp::min(item, min);
            sum += item;
        }

        sum -= max;
        sum -= min;
        (sum as f64) / ((len - 2) as f64)
    }
}